src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -404,13 +404,19 @@ class Global(var currentSettings: Settings, reporter0: Reporter)
     }
 
     def apply(unit: CompilationUnit): Unit
+    def pickleJava = true
+
+    // run only the phases needed
+    private val runThisPhaseForJava: Boolean = {
+      this.id > (if (pickleJava) currentRun.picklerPhase.id
+        else if (createJavadoc) currentRun.typerPhase.id
+        else currentRun.namerPhase.id)
```

Comment:
We don't want to run typer and refchecks on `.java` files (as this could emit warnings and errors that aren't our business to report, like reference to deprecated members), we just want to run  `pickler`.

Reply:
makes sense - updated to only run pickle